x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/shared/ml_inference/utils.ts

```diff
@@ -58,6 +58,13 @@ export const isSupportedMLModel = (model: TrainedModelConfigResponse): boolean =
   );
 };
 
+export const getModelTobeFiltered = (model: TrainedModelConfigResponse): string | undefined => {
+  if (
+    Object.keys(model.inference_config).some((k) => k === SUPPORTED_PYTORCH_TASKS.TEXT_EXPANSION)
```

Comment:
`inference_config` only has a [single property](https://www.elastic.co/guide/en/elasticsearch/reference/8.7/get-trained-models.html#ml-get-trained-models-results), which is the model task type; it's easier to check `model?.inference_config?.text_expansion`.